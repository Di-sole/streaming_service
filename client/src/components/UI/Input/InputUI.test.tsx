import { render, screen } from '@testing-library/react';
import InputUI from './InputUI';

jest.mock('next/router', () => ({
    useRouter() {
        return {
            pathname: '',
            asPath: '',
        };
    },
}));

describe('INPUTUI SNAPSHOTS TESTS', () => {
    test('type="search" InputUI should not be changed', () => {
        render(
            <InputUI
                type="search"
                inputType="search"
                placeholder="search"
                value="value"
                focus={false}
                onChange={() => 'value'}
            />,
        );
        const input = screen.getByTestId('input');
        expect(input).toMatchSnapshot();
    });

    test('type="email" InputUI should not be changed', () => {
        render(
            <InputUI
                type="email"
                inputType="email"
                placeholder="email"
                value="value"
                focus={false}
                onChange={() => 'value'}
            />,
        );
        const input = screen.getByTestId('input');
        expect(input).toMatchSnapshot();
    });

    test('type="password" InputUI should not be changed', () => {
        render(
            <InputUI
                type="password"
                inputType="password"
                placeholder="password"
                value="value"
                focus={false}
                onChange={() => 'value'}
            />,
        );
        const input = screen.getByTestId('input');
        expect(input).toMatchSnapshot();
    });

    test('type="comment" InputUI should not be changed', () => {
        render(
            <InputUI
                type="comment"
                inputType="text"
                placeholder="comment"
                value="value"
                focus={false}
                onChange={() => 'value'}
            />,
        );
        const input = screen.getByTestId('input');
        expect(input).toMatchSnapshot();
    });
});
