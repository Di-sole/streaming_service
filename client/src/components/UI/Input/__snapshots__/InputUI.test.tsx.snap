// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INPUTUI SNAPSHOTS TESTS type="comment" InputUI should not be changed 1`] = `
<div
  class="container container_comment container_focus"
  data-testid="input"
>
  <div
    class="container__placeholder"
  >
    comment
  </div>
  <input
    class="container__input"
    type="text"
    value="value"
  />
  <img
    alt="close"
    class="container__icon"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style="color: transparent;"
    width="40"
  />
</div>
`;

exports[`INPUTUI SNAPSHOTS TESTS type="email" InputUI should not be changed 1`] = `
<div
  class="container container_email container_focus"
  data-testid="input"
>
  <div
    class="container__placeholder"
  >
    email
  </div>
  <input
    class="container__input"
    type="email"
    value="value"
  />
  <img
    alt="user"
    class="container__icon"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style="color: transparent;"
    width="40"
  />
</div>
`;

exports[`INPUTUI SNAPSHOTS TESTS type="password" InputUI should not be changed 1`] = `
<div
  class="container container_password container_focus"
  data-testid="input"
>
  <div
    class="container__placeholder"
  >
    password
  </div>
  <input
    class="container__input"
    type="password"
    value="value"
  />
  <img
    alt=""
    class="container__icon container__icon_eye"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style="color: transparent;"
    width="40"
  />
</div>
`;

exports[`INPUTUI SNAPSHOTS TESTS type="search" InputUI should not be changed 1`] = `
<div
  class="container container_search container_focus"
  data-testid="input"
>
  <div
    class="container__placeholder"
  >
    search
  </div>
  <input
    class="container__input"
    type="search"
    value="value"
  />
  <img
    alt="close"
    class="container__icon"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style="color: transparent;"
    width="40"
  />
</div>
`;
